dist/logs/logzioLogsFunction/core.js

```diff
@@ -1,60 +1,186 @@
+        }
+        await Promise.all(promises);
+        if (!arrayIsEmpty(filesToUpload)) {
+            context.log('Uploaded logs to back up container.');
+        }
+    } catch (error) {
+        context.log('Error: ', error);
+    }
+};
+const writeEventToBlob = async (event, filesToUpload, foldersToDelete, context) => {
+    if (logIndex >= maxBatchSize) {
+        fileName = createFileName();
+        updateFolderSize();
+        logIndex = 1;
+    } else {
+        logIndex++;
+        const eventWithNewLine = JSON.stringify(event).concat('\n');
+        const concatFolderToFile = folderName + "\\" + fileName;
+        const filePath = workingDir + "\\" + concatFolderToFile;
+        try {
+            await appendFileAsync(filePath, eventWithNewLine);
+            pushFileToUploadAsync(filesToUpload, concatFolderToFile, context);
+            pushFolderToDeleteAsync(foldersToDelete, folderName, context);
+        }
+        catch (e) {
+            context.log(`Error was thrown in appendFile, ${e}`);
+        }
+    };
+};
+const pushFileToUploadAsync = (filesToUpload, filePath, context) => {
+    if (!filesToUpload.includes(filePath)) {
+        filesToUpload.push(filePath);
+    }
+}
+const pushFolderToDeleteAsync = (foldersToDelete, folderPath, context) => {
+    if (!foldersToDelete.includes(folderPath)) {
+        foldersToDelete.push(folderPath);
+    }
+}
 
-  const parseMessagesArray = dataParser.parseEventHubLogMessagesToArray(eventHubMessages);
-  context.log(`About to send ${parseMessagesArray.length} logs...`);
-  parseMessagesArray.forEach((log) => {
-    context.log(JSON.stringify(`logging: ${log}`));
-    logzioShipper.log(log);
-  });
+const updateFolderNameIfFull = () => {
+    if (folderSize >= MB) {
+        folderName = createFolderName();
+        folderSize = 0;
+    }
+}
 
-  logzioShipper.sendAndClose(callBackFunction);
-};
+module.exports = async function processEventHubMessages(context, eventHubs) {
+    createFileName(context);
+    const filesToUpload = [];
+    const foldersToDelete = [ folderName ];
+    const callBackFunction = getCallBackFunction(context);
+    const { host, token, bufferSize, timeout } = getParserOptions();
+    const { storageConnectionString, containerName } = getContainerDetails();
+    const containerClient = new ContainerClient(storageConnectionString, containerName);
+    const logzioShipper = logger.createLogger({
+        token,
+        host,
+        type: 'eventHub',
+        protocol: 'https',
+        internalLogger: context,
+        compress: true,
+        debug: true,
+        callback: callBackFunction,
+        bufferSize: bufferSize || 100,
+        timeout: timeout || 180000,
+    });
+    eventHubs[0].records.map(async eventHub => {
+        updateTimestamp(eventHub);
+        if (process.env.ParseEmptyFileds == 'true') {
+            deleteEmptyFields(eventHub);
+        }
+        try {
+            logzioShipper.log(eventHub);
```

Comment:
eventHub is a single log? line? maybe you should change it in the future to send a bulk of those somehow

Reply:
the shipper handles bulking for me